### ISSUE TITLE
State property is not defined for ‘Viewport Selection’ button.

### ISSUE THREAD

**Issue by azaslonov** [paperbits/paperbits-demo#84]
User Experience:

User would find difficulty to identify state of the controls if state property  are not defined for any control as they won't get the context of that control and also they won't be able to identify current state of control.

Note: User credentials should NOT be included in the bug.

 

Repro Steps:

1. Open Azure portal with valid credentials and create a API Management.

2. Click Developer Portal button to navigate to the portal.

3. Wait portal to load in design mode.

4. Open Developer tool using F12 key and navigate to the bottom nav and select 'Viewport Selection' button.

5. Verify whether state property is defined or not for 'Viewport Selection' button.

 

Actual Result:

State property as 'Expanded' and 'Collapsed' is not defined for ‘Viewport Selection’ button, due to this screen reader does not announce state as 'Expanded' and 'Collapsed' after activating 'Viewport Selection' button.

 

Expected Result:

State property as 'Expanded' and 'Collapsed' should be defined for ‘Viewport Selection’ button, so that screen reader can announce state as 'Expanded' and 'Collapsed' for ‘Viewport Selection’ button.

Note:

Similar issue exist with Chrome.

Similar issue exist for control: Role selection.

 

MAS Reference:

MAS 4.1.2 - Name, Role, Value

 

Reference Links

Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information please use this Link. “HCL Staff should not log any third party external bugs. Should be routed to edad team”.
Please reach out to C&AI Teams channel for any process related queries.

![image](https://user-images.githubusercontent.com/2320302/76664855-edcb6c00-6542-11ea-95a4-70d8f22c36de.png)

**Comment by azaslonov** [comment]
Fixed.
